### PR TITLE
remove matching / JuMP requirement

### PR DESCRIPTION
**Pull request by sbromberger**
per #389

### CODE REVIEW

**Comment by sbromberger** [comment]
@jpfairbanks @CarloLucibello I am planning on merging this and tagging a new version soon (within the next couple of days). Please let me know if you have any strong objections.

File: README.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ symmetric difference, blkdiag, induced subgraphs, products (cartesian/scalar)
 
 - **flow operations:** maximum flow
 
-- **matching:** bipartite maximum matching
+- **matching:** Matching functions have been moved to [LG-matching.jl](https://github.com/JuliaGraphs/LG-matching.jl).
```
Comment (tkelman) [inline README.md]: no dash in the name, that wouldn't have worked as a module identifier

Comment (sbromberger) [inline README.md]: Good catch. It was originally "LG-matching" and of course it didn't work, so I removed the dashes - everywhere but here.

**Comment by CarloLucibello** [comment]
I would keep the documentation here and just add a few lines suggesting to install LGMatching
JuliaGraphs/LGmatching.jl#1

**Pull request merged**